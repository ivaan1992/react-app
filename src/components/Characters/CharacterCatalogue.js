import React from 'react';
import PropTypes from 'prop-types';
import './Characters.css';

const CharacterCatalogue = ({
  img, name, status, location,
}) => (
  <div className="d-flex card-catalogue">
    <img src={img} alt={name} />
    <div className="info-character-catalogue">
      <h1 className="character-name-cat">{name}</h1>
      {(
        () => {
          if (status === 'Dead') {
            return (
              <div className="status-red">{status}</div>
            );
          } if (status === 'Alive') {
            return (
              <div className="status-green">{status}</div>
            );
          } return (
            <div className="status-gray">{status}</div>
          );
        }
      )()}
      <h3>{location.name}</h3>
    </div>
  </div>
);

CharacterCatalogue.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
};

export default CharacterCatalogue;
