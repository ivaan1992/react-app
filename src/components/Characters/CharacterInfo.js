import React from 'react';
import PropTypes from 'prop-types';

const CharacterInfo = ({
  img, name, gender, status, location, species, type,
}) => (
  <div>
    <img src={img} alt={name} />
    <div className="info">
      <h1>{name}</h1>
      <p>{gender}</p>
      <p>{status}</p>
      <p>{location.name}</p>
      <p>{species}</p>
      <p>{type}</p>
    </div>
  </div>
);

CharacterInfo.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default CharacterInfo;
