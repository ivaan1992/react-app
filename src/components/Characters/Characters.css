.d-flex {
  display: flex;
}

.chars-container {
  width: 100%;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  margin: 0 auto;
  background-color: #24282f;
}

.chars-cards {
  padding-top: 3rem;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  gap: 3rem;
  width: 80%;
  margin: 0 auto;
}

.card-catalogue {
  width: 80%;
  margin: 0 auto;
  margin-bottom: 3rem;
  border-radius: 1rem;
  background: #3c3e44;
}

.card-catalogue img {
  width: 50%;
  height: 25vh;
  border-top-left-radius: 1rem;
  border-bottom-left-radius: 1rem;
  cursor: pointer;
}

.info-character-catalogue {
  padding-left: 2rem;
  margin-top: 3rem;
  width: 50%;
}

.white {
  color: #fff;
}

.character-name-cat {
  width: 80%;
  font-size: 35px;
  margin-bottom: 1rem;
}

.character-name-cat:hover {
  color: #fa9805;
  cursor: pointer;
}

.status-red {
  width: 6rem;
  height: 2vh;
  border-radius: 2rem;
  text-align: center;
  background: #af1515;
  font-weight: bold;
  font-size: 1.25rem;
  margin-bottom: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.status-green {
  width: 6rem;
  height: 2vh;
  border-radius: 2rem;
  text-align: center;
  background: #008000;
  font-weight: bold;
  font-size: 1.25rem;
  margin-bottom: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.status-gray {
  width: 6rem;
  height: 2vh;
  border-radius: 2rem;
  text-align: center;
  background: #000;
  font-weight: bold;
  font-size: 1.25rem;
  margin-bottom: 2rem;
  display: flex;
  justify-content: center;
  align-items: center;
}
