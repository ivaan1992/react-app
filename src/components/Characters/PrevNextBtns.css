.d-flex {
  display: flex;
}

.prev-next-btns {
  justify-content: center;
  align-items: center;
  padding-top: 3rem;
}

.prev-next-btn {
  width: 5rem;
  height: 5vh;
  margin-right: 2rem;
  background: #fa9805;
  color: #fff;
  font-size: 0.75rem;
  font-weight: 800;
  border: none;
}

.prev-next-btn:hover {
  background: #1367bb;
}
