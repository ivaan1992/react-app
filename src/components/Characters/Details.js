import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { getCharacter } from '../../redux/Characters/CharactersRedux';

const Details = ({ isVisible, onClickEvent }) => {
  const dispatch = useDispatch();
  const {
    img, name, gender, location, species, type, status,
  } = useSelector((state) => state.characterReducer.details);

  useEffect(() => {
    getCharacter(dispatch);
  }, []);

  return (
    <>
      { isVisible ? (
        <div>
          <button onClick={onClickEvent} type="button">CLOSE</button>
          <img src={img} alt={name} />
          <div className="info">
            <h1>{name}</h1>
            <p>{gender}</p>
            <p>{status}</p>
            <p>{location.name}</p>
            <p>{species}</p>
            <p>{type}</p>
          </div>
        </div>

      )
        : (
          <></>
        )}
    </>
  );
};

export default Details;

Details.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onClickEvent: PropTypes.func.isRequired,
};
