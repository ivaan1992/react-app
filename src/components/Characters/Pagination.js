import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { getCharacter } from '../../redux/Characters/CharactersRedux';
import './Pagination.css';

const Pagination = () => {
  const [page, setPage] = useState(1);
  const dispatch = useDispatch();

  useEffect(() => {
    getCharacter(page, dispatch);
  }, [page]);

  return (
    <div className=" d-flex prev-next-btns">
      { page
        ? <button className="prev-next-btn" type="button" onClick={() => setPage(page - 1)}>Previous</button>
        : '' }
      <button className="prev-next-btn" type="button" onClick={() => setPage(page + 1)}>Next</button>
    </div>
  );
};
export default Pagination;
