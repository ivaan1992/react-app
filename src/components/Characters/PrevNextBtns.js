import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { getCharacter } from '../../redux/Characters/CharactersRedux';
import './PrevNextBtns.css';

const PrevNextBtns = () => {
  const [page, setPage] = useState(1);
  const dispatch = useDispatch();

  useEffect(() => {
    getCharacter(page, dispatch);
  }, [page]);

  return (
    <div className=" d-flex prev-next-btns">
      { page
        ? <button className="prev-next-btn" type="button" onClick={() => setPage(page - 1)}>Previous</button>
        : '' }
      <button className="prev-next-btn" type="button" onClick={() => setPage(page + 1)}>Next</button>
    </div>
  );
};
export default PrevNextBtns;
