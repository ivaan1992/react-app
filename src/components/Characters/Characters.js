import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CharacterCatalogue from './CharacterCatalogue';
import PrevNextBtns from './PrevNextBtns';
import SearchBar from '../Search/SearchBar';
import './Characters.css';
import Details from './Details';
import { setSelectedCharacter } from '../../redux/Characters/CharactersRedux';

const Characters = () => {
  const chars = useSelector((state) => state.characterReducer.chars);
  const [isSelected, setIsSelected] = useState(false);
  const dispatch = useDispatch();

  const handleSelectedChar = (selectedChar) => {
    setIsSelected(true);
    setSelectedCharacter({ character: selectedChar, dispatch });
  };

  const handleCloseSelectedChar = () => {
    setIsSelected(false);
    setSelectedCharacter({ character: {}, dispatch });
  };

  return (
    <div className="d-flex chars-container">
      <Details isVisible={isSelected} onClickEvent={handleCloseSelectedChar} />
      <SearchBar />
      <div className="chars-cards">
        {
            chars.map(({
              name, gender, status, image, id, location, species, type,
            }) => (
              <CharacterCatalogue
                key={id}
                name={name}
                gender={gender}
                status={status}
                img={image}
                location={location}
                species={species}
                type={type}
                onClickEvent={handleSelectedChar}
              />
            ))
        }
      </div>
      <PrevNextBtns />
    </div>
  );
};

export default Characters;
