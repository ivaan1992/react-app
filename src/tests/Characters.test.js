import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Characters from '../components/Characters/Characters';

it('Testing Characters', () => {
  const list = renderer
    .create(
      <Provider store={store}>
        <Characters />
      </Provider>,
    )
    .toJSON();
  expect(list).toMatchSnapshot();
});
