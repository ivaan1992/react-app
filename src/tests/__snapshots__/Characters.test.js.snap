// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Characters 1`] = `
<div
  className="d-flex chars-container"
>
  <form
    className="search-char-form"
    onSubmit={[Function]}
  >
    <input
      className="search-bar-char"
      name="name"
      onChange={[Function]}
      placeholder="Search by Name"
    />
    <button
      className="search-bar-name"
      type="submit"
    >
      Search
    </button>
  </form>
  <div
    className="chars-cards"
  />
  <div
    className=" d-flex prev-next-btns"
  >
    <button
      className="prev-next-btn"
      onClick={[Function]}
      type="button"
    >
      Previous
    </button>
    <button
      className="prev-next-btn"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
